### DISCUSSION TITLE
Support for different opacity text

### DISCUSSION THREAD

**Discussion by crimist** [wezterm/wezterm#1392] [Q&A] [answered]
Hi, I'm not sure exactly how to word the issue so I'll descibe it with images attached:

Windows Terminal:
![image](https://user-images.githubusercontent.com/20033376/145736300-41ac123c-66e5-4029-8908-a8f04c189a2e.png)

WezTerm:
![image](https://user-images.githubusercontent.com/20033376/145736304-6f348e08-694b-4895-b368-1eaaaa3e14ca.png)

WezTerm with `foreground = "#f8f8f2"` added to the configuration file:
![image](https://user-images.githubusercontent.com/20033376/145736316-07a4a4cb-f38a-4c5c-aa40-7f1bce4cccc3.png)

I use oh-my-zsh which supports suggestions in the form of commands that were previously ran. In this case it is suggesting to run `echo "Example"` after typing `echo`.

Windows terminal correctly renders the autocompletion preview for the suggestion as a ligher/less opaque font when compared to the default foreground font. However WezTerm seems to render it with the same font weight and opacity as the foreground. You can see the behavior most clearly in the last image where i set the foreground to near white.

Is there a way to configure WezTerm such that the autocompletion preview displays the desired font weight/opacity similar to how Windows Terminal does?

---

On a slightly different note just wanna say this is a fantastic project. I've been looking for a GPU rendered windows terminal emu for a while and this does everything I need and more. The configuration system is also far more effective and understandable than anything else I've tried. Great work!

**Comment by wez** [comment, marked as answer]
That sort of hint/preview in line editors is typically something like a foreground attribute of bold+black which tends to be a dark grey in color schemes that are closer to the ANSI terminal palette.  Does your chosen color scheme actually give a useable spread of the 16 color palette?  Some color schemes are fairly monochrome and don't have a lot of distinction between colors 0-7 and 8-15.

The `TERM` and `COLORTERM` environment variables influence how shells and other console programs choose to output text (especially with colors), so I would suggest comparing what is set in your Windows Terminal with what is set in WezTerm.

Sometimes, the half-bright or dim attribute is also used for that kind of hint/preview.  Wezterm has a default set of [font_rules](https://wezfurlong.org/wezterm/config/lua/config/font_rules.html) that will select a lighter weight version of your font (if available), or that will adjust the brightness of the text if no lighter weight font is available.

You can use `wezterm ls-fonts` to see an explanation of which fonts and which treatment will be used for your configuration.

**Reply by dhruvkb** [reply]
Referencing #6026, is it possible to customise `font_rules` to forcefully go for reduced opacity for half-intensity/dim text even when a lighter weight is available?

**Comment by marekkaczkowski** [comment]
@wez is it possible to change the hint color like `hint_fg`?
